examples/99-advanced/openfoam-cooling.py

```diff
@@ -0,0 +1,173 @@
+# cooled down by a fan, while the ``air`` dataset is the air, containing
+# several scalar arrays including the velocity and temperature of the air.
+
+structure, air = examples.download_electronics_cooling()
+structure, air
+
+
+###############################################################################
+# Plot the Electronics
+# ~~~~~~~~~~~~~~~~~~~~
+# Here we plot the temperature of the electronics using the ``"reds"`` colormap
+# and improve the look of the plot using surface space ambient occlusion with
+# :func:`enable_ssao() <pyvista.Plotter.enable_ssao>`.
+
+pl = pv.Plotter()
+pl.enable_ssao(radius=0.01)
+pl.add_mesh(
+    structure, scalars='T', smooth_shading=True, split_sharp_edges=True, cmap='reds', ambient=0.2
+)
+pl.enable_anti_aliasing('fxaa')  # also try 'ssaa'
+pl.show()
+
+
+###############################################################################
+# Plot Air Velocity
+# ~~~~~~~~~~~~~~~~~
+# Let's plot the velocity of the air.
+#
+# Start by clipping the air dataset with :func:`clip()
+# <pyvista.DataSetFilters.clip>` and plotting it alongside the electronics.
+#
+# As you can see, the air enters from the front of the case (left) and is being
+# pushed out of the "back" of the case via a fan.
+
+# Clip the air in the XY plane
+z_slice = air.clip('z', value=-0.005)
+
+# Plot it
+pl = pv.Plotter()
+pl.enable_ssao(radius=0.01)
+pl.add_mesh(z_slice, scalars='U', lighting=False, scalar_bar_args={'title': 'Velocity'})
+pl.add_mesh(structure, color='w', smooth_shading=True, split_sharp_edges=True)
+pl.camera_position = 'xy'
+pl.camera.roll = 90
+pl.enable_anti_aliasing('fxaa')
+pl.show()
+
+
+###############################################################################
+# Plot Air Temperature
+# ~~~~~~~~~~~~~~~~~~~~
+# Let's also plot the temperature of the air. This time, let's also plot the
+# temperature of the components.
+
+pl = pv.Plotter()
+pl.enable_ssao(radius=0.01)
+pl.add_mesh(
+    z_slice, scalars='T', lighting=False, scalar_bar_args={'title': 'Temperature'}, cmap='reds'
+)
+pl.add_mesh(
+    structure,
+    scalars='T',
+    smooth_shading=True,
+    split_sharp_edges=True,
+    cmap='reds',
+    show_scalar_bar=False,
+)
+pl.camera_position = 'xy'
+pl.camera.roll = 90
+pl.enable_anti_aliasing('fxaa')
+pl.show()
+
+
+###############################################################################
+# Plot Steamlines - Flow Velocity
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Now, let's plot the steamlines of this dataset so we can see how the air is
+# flowing through the case.
+#
+# Generate streamlines using :func:`streamlines() <pyvista.DataSetFilters.streamlines>`.
+
+# Have our streamlines start from the regular openings of the case.
+lines = []
+z_rng = np.linspace(0, 0.03, 5)
+for x in np.linspace(0.045, 0.105, 7, endpoint=True):
+    lines.extend([air.streamlines(start_position=[x, 0.2, z], max_time=2.0) for z in z_rng])
+lines = pv.merge(lines)
```

Comment:
Instead of generating this way, it is more vtk-onic to make a PointSet or PolyData with all the source points and do `streamlines_from_source`.  Then no merge is needed.  But this could also just be commented on to give hints of multiple ways to accomplish this.

Reply:
Resolved in 2a78fa8aede41f258650f2a7514d295763d43ac8.